### PR TITLE
docs: enrich stand alone mode's doc

### LINKED ISSUE #1597: request help: Better documentation for standalone mode configuration
### Issue description
The documentation for standalone mode is very few, it doesn't cover many scenarios, like host match, plugins, people can only make trials with lots of failures. It is time-consuming. Please complete the document for this.

### Environment

* apisix version (cmd: `apisix version`):
* OS:

### LINKED ISSUE #2635: apisix stand-alone mode
### Issue description
我突然看到config_center可以配置为yaml,请问哪有apisix.config的详细编写文档，我目前使用到的插件是jwt-auth

### Environment

* apisix version (cmd: `apisix version`): 1.2
* OS:

### PR DESCRIPTION
**Pull request by spacewander**
Fix #1597
Fix #2635

Signed-off-by: spacewander <spacewanderlzx@gmail.com>

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [x] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: doc/stand-alone.md [hunk 1]
```diff
@@ -228,3 +228,70 @@ ssl:
             - "yourdomain.com"
 #END
 ```
+
+#### How to config global rule
```
Comment (juzhiyuan) [inline doc/stand-alone.md:232]: `configure` is a variable :)

Comment (juzhiyuan) [inline doc/stand-alone.md:232]:
```suggestion
#### How to configure a global rule
```

File: doc/stand-alone.md [hunk 2]
```diff
@@ -228,3 +228,70 @@ ssl:
...
+routes:
+  -
+    uri: /hello
+    upstream_id: 1
+    plugins:
+        http-logger:
+            batch_max_size: 1
+            uri: http://127.0.0.1:1980/log
+plugin_metadata:
+  - id: http-logger # note the id is the plugin name
```
Comment (juzhiyuan) [inline doc/stand-alone.md:271]: Do we have clear documentation about this tip?

**Review by juzhiyuan — APPROVED** [review]
Others look good to me

File: doc/stand-alone.md (hunk 2 above)
Comment (spacewander) [reply doc/stand-alone.md:271]: @juzhiyuan 
This one.

Comment (juzhiyuan) [reply doc/stand-alone.md:271]: I find it here, why not use `plugin_name` like here[1]?

[1] https://github.com/apache/apisix/blob/master/doc/admin-api.md#plugin-metadata

**Review by juzhiyuan — APPROVED** [review]

**Review by membphis — APPROVED** [review]

**Comment by Firstsawyou** [comment]
LGTM.

**Pull request merged**